packages/backend/src/remote/activitypub/models/tag.ts

```diff
@@ -16,3 +16,34 @@ export function extractApHashtagObjects(tags: IObject | IObject[] | null | undef
 	if (tags == null) return [];
 	return toArray(tags).filter(isHashtag);
 }
+
+export function extractQuoteUrl(tags: IObject | IObject[] | null | undefined): string | null {
+	if (tags == null) return null;
+
+	let quotes: ILink[] = toArray(tags)
+		.filter(isLink)
+		.filter(link => link.mediaType === 'application/activity+json');
+
+	// Multiple quotes are not supported, try to reduce to only 1.
+	if (quotes.length > 1) {
+		// Check if there maybe is one intended for Misskey.
+		quotes = quotes.filter(link => toArray(link.rel).includes('https://misskey-hub.net/ns#_misskey_quote'));
+	}
+	if (quotes.length > 1) {
+		// Deduplicate by href value.
+		const hrefs = new Set();
+		quotes = quotes.filter(link => {
+			if (hrefs.has(link.href)) return false;
+
+			hrefs.add(link.href);
+			return true;
+		});
+	}
+
+	if (quotes.length === 1) {
+		return quotes.href;
+	} else {
+		// either multiple quotes (unsupported) or no quote
+		return null;
+	}
```

Comment:
Other servers may allow multiple quotes per post. Perhaps it would be better to take the first link from the array if `quotes.length >= 1`?